File: OAiP_1_semestr/Lab12/Stack.cpp
#include "Stack.h"
Stack* first(char d) {
	Stack* pv = new Stack;
	pv->d = d;
	pv->p = 0;
	return pv;
}
void add(Stack** top, char d) {
	Stack* pv = new Stack;
	pv->d = d;
	pv->p = *top;
	*top = pv;
}
char izv(Stack** top) {
	char ud = (*top)->d;
	Stack* pv = *top;
	*top = (*top)->p;
	delete pv;
	return ud;
}

File: TVSBD_1_semestr/Lab4/Lab4_1/pack/Stack.java
package pack;

public class Stack<Type> {

    private int size;
    private Type[] array;
    private int top;
 
    public Stack(int size) {
        this.size = size;
        array = (Type[]) new Object[size];
        top = -1;
    }

    public void push(Type element) {
        if(top == size - 1) {
            return;
        } else {
            array[++top] = element;
        }
    }

    public Type pop() {
        return array[top--];
    }

    public Type safePop() {
        return array[top];
    }

    public boolean isEmpty() {
        return (top == -1);
    }

    public boolean isFull() {
        return (top == size - 1);
    }
}
